packages/map-toolkit/src/deckgl/viewport-scale/utils.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2025 Hypergiant Galactic Systems Inc. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { distance } from '@turf/distance';
+import { UNIT_MAP } from './constants';
+import type { GetViewportScaleArgs } from './types';
+
+const numberFormatter = Intl.NumberFormat('en-US');
+
+export function getViewportScale({
+  bounds,
+  unit,
+  formatter = numberFormatter,
+}: GetViewportScaleArgs) {
+  if (!bounds) {
+    return '-- x --' as const;
```

Comment:
should we update the placeholder text to be this value instead of ### x ###?

if so, see my comment above for an alternative in the render

also, I think we should include the unit.toUpperCase() in this fallback text?


```suggestion
    return `-- x -- ${unit.toUpperCase()}` as const;
```